pandas/core/frame.py

```diff
@@ -3481,11 +3481,12 @@ def __getitem__(self, key):
         if is_hashable(key) and not is_iterator(key):
             # is_iterator to exclude generator e.g. test_getitem_listlike
             # shortcut if the key is in columns
-            if self.columns.is_unique and key in self.columns:
-                if isinstance(self.columns, MultiIndex):
-                    return self._getitem_multilevel(key)
+            is_mi = isinstance(self.columns, MultiIndex)
+            if not is_mi and key in self.columns.drop_duplicates(keep=False):
```

Comment:
comment here pointing back to the relevant issue?

Reply:
Added